### PR TITLE
Remove SVN revision from all version strings and numbers and add isodate instead

### PR DESCRIPTION
**Pull request by frosch123**
Change: Non-tag revisions are now named '<commitdate>-<branch>-g<shorthash><modified>'.
Remove: [NewGRF] OpenTTD version no longer contains a revision number.

### CODE REVIEW

**Comment by frosch123** [comment]
MSVB determineversion.vbs is completely untested :)

**Comment by TrueBrain** [comment]
Commit message is weird. Like there are 2 commits in 1 commit message?

**Comment by frosch123** [comment]
You mean the last hunk of rev.cpp.in should be a separate commit for the second line of the commit message?

File: Makefile.bundle.in [hunk 1]
```diff
@@ -12,15 +12,15 @@
...
 VERSIONS := $(shell AWK="$(AWK)" "$(ROOT_DIR)/findversion.sh")
-REV      := $(shell echo "$(VERSIONS)" | cut -f 1 -d'	')
+NAME     := $(shell echo "$(VERSIONS)" | cut -f 1)
 
-# Make sure we have something in REV
-ifeq ($(REV),)
-REV := norev000
+# Make sure we have something in NAME
+ifeq ($(NAME),)
+NAME := norev000
```
Comment (TrueBrain) [inline Makefile.bundle.in]: is 'norev000' a smart idea? Maybe 'nameless'? 'noname'?

File: Makefile.src.in [hunk 2]
```diff
@@ -293,10 +293,10 @@ $(ENDIAN_CHECK): $(SRC_DIR)/endian_check.cpp
 # Revision files
 
 $(SRC_DIR)/rev.cpp: $(CONFIG_CACHE_VERSION) $(SRC_DIR)/rev.cpp.in
-	$(Q)cat $(SRC_DIR)/rev.cpp.in      | sed "s@\!\!REVISION\!\!@$(REV_NR)@g;s@!!VERSION!!@$(REV)@g;s@!!MODIFIED!!@$(MODIFIED)@g;s@!!DATE!!@`date +%d.%m.%y`@g" > $(SRC_DIR)/rev.cpp
+	$(Q)cat $(SRC_DIR)/rev.cpp.in      | sed "s@\!\!ISODATE\!\!@$(ISODATE)@g;s@!!VERSION!!@$(NAME)@g;s@!!MODIFIED!!@$(MODIFIED)@g;s@!!DATE!!@`date +%d.%m.%y`@g" > $(SRC_DIR)/rev.cpp
 
 $(SRC_DIR)/os/windows/ottdres.rc: $(CONFIG_CACHE_VERSION) $(SRC_DIR)/os/windows/ottdres.rc.in
-	$(Q)cat $(SRC_DIR)/os/windows/ottdres.rc.in | sed "s@\!\!REVISION\!\!@$(REV_NR)@g;s@!!VERSION!!@$(REV)@g;s@!!DATE!!@`date +%d.%m.%y`@g" > $(SRC_DIR)/os/windows/ottdres.rc
+	$(Q)cat $(SRC_DIR)/os/windows/ottdres.rc.in | sed "s@\!\!ISODATE\!\!@$(ISODATE)@g;s@!!VERSION!!@$(NAME)@g;s@!!DATE!!@`date +%d.%m.%y`@g" > $(SRC_DIR)/os/windows/ottdres.rc
```
Comment (TrueBrain) [inline Makefile.src.in]: !!VERSION!! is replaced with $(name). Feels a bit weird?

File: findversion.sh [hunk 3]
```diff
@@ -70,57 +61,29 @@ ROOT_DIR=`pwd`
...
+	else
+		NAME="${ISODATE}-${BRANCH}-g${SHORTHASH}"
 	fi
-	REV_NR=`LC_ALL=C HGPLAIN= hg log -f -k "(svn r" -l 1 --template "{desc|firstline}\n" | grep "^(svn r[0-9]*)" | sed "s@.*(svn r\([0-9]*\)).*@\1@"`
-	if [ -z "$REV_NR" ]; then
-		# No rev? Maybe it is a custom hgsubversion clone
-		REV_NR=`LC_ALL=C HGPLAIN= hg parent --template="{svnrev}"`
+
+	if [ "$MODIFIED" -eq "2" ]; then
+		NAME="${NAME}M"
```
Comment (TrueBrain) [inline findversion.sh]: more common this is called '-dirty'?

File: projects/determineversion.vbs [hunk 4]
```diff
@@ -21,312 +21,132 @@ Sub FindReplaceInFile(filename, to_find, replacement)
 	file.Close
 End Sub
 
-Sub UpdateFile(modified, revision, version, cur_date, filename)
+Sub UpdateFile(modified, isodate, version, cur_date, filename)
 	FSO.CopyFile filename & ".in", filename
 	FindReplaceInFile filename, "!!MODIFIED!!", modified
-	FindReplaceInFile filename, "!!REVISION!!", revision
+	FindReplaceInFile filename, "!!ISODATE!!", isodate
 	FindReplaceInFile filename, "!!VERSION!!", version
```
Comment (TrueBrain) [inline projects/determineversion.vbs:28]: Version become 'name', not?

File: src/rev.cpp.in [hunk 5]
```diff
@@ -27,17 +27,17 @@ bool IsReleasedVersion()
 /**
  * The text version of OpenTTD's revision.
  * This will be either "<major>.<minor>.<build>[-RC<rc>]",
- * "r<revision number>[M][-<branch>]" or "norev000".
+ * "<hash>[M][-<branch>]" or "norev000".
```
Comment (TrueBrain) [inline src/rev.cpp.in]: This is not true, is it? It is <isodate>-<branch>-<hash>M?

File: src/rev.cpp.in [hunk 6]
```diff
@@ -27,17 +27,17 @@ bool IsReleasedVersion()
...
  *
  * The major, minor and build are the numbers that describe releases of
  * OpenTTD (like 0.5.3). "-RC" is used to flag release candidates.
  *
- * The revision number is fairly straight forward. The M is to show that
+ * The hash is fairly straight forward. The M is to show that
```
Comment (TrueBrain) [inline src/rev.cpp.in]: This sentence no longer makes sense; is a hash straight forward? :)

**Comment by TrueBrain** [comment]
Didn't tried to understand the vbs; that is for @glx22  :D

**Comment by frosch123** [comment]
All scripts now consistently use "VERSION" and "ISODATE", no more "NAME" or "REVISION".

Changing "norev000" and "M" for modified is out of scope of this patch, it keeps the old behaviour here.

**Comment by TrueBrain** [comment]
Looks good to me! Up to our Windows expert :)

**Review by glx22 — COMMENTED** [review]
[SRC] Compiling rev.cpp
d:/developpement/GitHub/OpenTTD/src/rev.cpp:58:55: error: 'gbe3cfc21' was not declared in this scope
 const byte _openttd_revision_modified = 20180411-6702-gbe3cfc21 20180411 0 be3cfc2143aab4f2dd04604c1377d80ba118073b;

File: projects/determineversion.vbs [hunk 7]
```diff
@@ -21,312 +21,132 @@ Sub FindReplaceInFile(filename, to_find, replacement)
...
 		Set rev_file = FSO.OpenTextFile("../.ottdrev", 1, True, 0)
-		DetermineSVNVersion = rev_file.ReadLine()
+		DetermineVersion = rev_file.ReadLine()
 		rev_file.Close()
+	ElseIf hash = "" Then
+		DetermineVersion = "norev000"
 	Else
-		If modified = 2 Then
-			version = version & "M"
+		Dim version
```
Comment (glx22) [inline projects/determineversion.vbs:136]: Redeclaration of variable version so remove this line

File: Makefile.bundle.in [hunk 8]
```diff
@@ -12,15 +12,15 @@
 # The revision is needed for the bundle name and creating an OSX application bundle.
 # Detect the revision
 VERSIONS := $(shell AWK="$(AWK)" "$(ROOT_DIR)/findversion.sh")
-REV      := $(shell echo "$(VERSIONS)" | cut -f 1 -d'	')
+VERSION  := $(shell echo "$(VERSIONS)" | cut -f 1)
```
Comment (glx22) [inline Makefile.bundle.in]: keep the -d for mingw

File: Makefile.src.in [hunk 9]
```diff
@@ -91,18 +91,18 @@ $(LANG_OBJS_DIR)/table/strings.h: $(LANG_DIR)/english.txt $(LANG_OBJS_DIR)/$(STR
 # Always run version detection, so we always have an accurate modified
 # flag
 VERSIONS := $(shell AWK="$(AWK)" "$(ROOT_DIR)/findversion.sh")
-MODIFIED := $(shell echo "$(VERSIONS)" | cut -f 3 -d'	')
+MODIFIED := $(shell echo "$(VERSIONS)" | cut -f 3)
```
Comment (glx22) [inline Makefile.src.in]: keep -d for mingw

File: Makefile.src.in [hunk 10]
```diff
@@ -91,18 +91,18 @@ $(LANG_OBJS_DIR)/table/strings.h: $(LANG_DIR)/english.txt $(LANG_OBJS_DIR)/$(STR
...
 
 # Use autodetected revisions
-REV      := $(shell echo "$(VERSIONS)" | cut -f 1 -d'	')
-REV_NR   := $(shell echo "$(VERSIONS)" | cut -f 2 -d'	')
+VERSION  := $(shell echo "$(VERSIONS)" | cut -f 1)
```
Comment (glx22) [inline Makefile.src.in]: keep -d for mingw

File: Makefile.src.in [hunk 11]
```diff
@@ -91,18 +91,18 @@ $(LANG_OBJS_DIR)/table/strings.h: $(LANG_DIR)/english.txt $(LANG_OBJS_DIR)/$(STR
...
+ISODATE  := $(shell echo "$(VERSIONS)" | cut -f 2)
```
Comment (glx22) [inline Makefile.src.in]: keep -d for mingw

File: findversion.sh [hunk 12]
```diff
@@ -129,19 +90,12 @@ elif [ -f "$ROOT_DIR/.ottdrev" ]; then
...
-
-if [ -n "$BRANCH" ]; then
-	REV="${REV}-$BRANCH"
+	ISODATE=""
+	TAG=""
+	VERSION=""
 fi
 
-echo "$REV	$REV_NR	$MODIFIED	$CLEAN_REV"
+echo "$VERSION\t$ISODATE\t$MODIFIED\t$HASH"
```
Comment (glx22) [inline findversion.sh]: use real tabs for mingw

**Comment by glx22** [comment]
With my changes it works correctly for mingw and MSVC

**Review by glx22 — APPROVED** [review]
it's ok for me

**Pull request merged**